Sources/Host/POSIX/PTrace.cpp

```diff
@@ -125,6 +125,25 @@ ErrorCode PTrace::resume(ProcessThreadId const &ptid, ProcessInfo const &pinfo,
   return doStepResume(false, ptid, signal, address);
 }
 
+ErrorCode PTrace::getClonedTID(ProcessId pid, unsigned long &clonedTID) {
```

Comment:
This is really sketchy because `PTRACE_GETEVENTMSG` is used for more than getting just the clone TID. You might wanna name it `getEventMessage()` and let the caller interpret the result (because the caller knows what the last ptrace event was and can interpret the result correctly).

Reply:
I don't think we should have a "high level" method mapping to such a generic low level function. I think it would be better to have multiple methods that do some high level task such as `getClonedTID` that all map to the same `ptrace(PTRACE_GETEVENTMSG,...)` call. Hence the name of this method. This method is only to be used to get the TID of a cloned(should be spawned?) thread.

Reply:
I would agree with you, except for the fact that what you are describing is not possible. `PTRACE_GETEVENTMSG` does different things depending on what the last event that occurred was.

Like in your proposal, all the methods (like `getClonedTID()`) would have the exact same code: a simple call to `ptrace(PTRACE_GETEVENTMSG, ...)`.